src/components/Login/LoginForm.js

```diff
@@ -68,18 +64,20 @@ export default class LoginModal extends React.Component {
           default:
             newMsg = `Failed to sign in: ${err.code}`;
           }
-          this.setState({ errorMsg: newMsg, waiting: false });
+          setErrorMsg(newMsg);
+          setWaiting(false);
         });
     } else {
-      this.setState({ waiting: false, errorMsg: 'Failed to reach Firebase login services' });
+      setWaiting(false);
+      setErrorMsg('Failed to reach Firebase login services');
     }
   };
 
-  updateUsername = (username) => this.setState({ username });
+  const updateUsername = (myUsername) => setUsername(myUsername);
```

Comment:
These 2 functions can be removed, as they do the exact same thing as the set functions. For example, `updateUsername` can be removed and all its calls can be switched to `setUsername`